src/test/java/com/thealgorithms/datastructures/lists/ReverseKGroupTest.java

```diff
@@ -0,0 +1,64 @@
+package com.thealgorithms.datastructures.lists;
+
+/**
+ * Test cases for Reverse K Group LinkedList
+ * Author: Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
+ */
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+public class ReverseKGroupTest {
+
+    @Test
+    public void testReverseKGroupWithEmptyList() {
+        ReverseKGroup reverser = new ReverseKGroup();
+        assertNull(reverser.reverseKGroup(null, 3));
+    }
+
+    @Test
+    public void testReverseKGroupWithSingleNodeList() {
+        ReverseKGroup reverser = new ReverseKGroup();
+        Node singleNode = new Node(5);
+        Node result = reverser.reverseKGroup(singleNode, 2);
+        assertEquals(5, result.value);
+        assertNull(result.next);
+    }
+
+    @Test
+    public void testReverseKGroupWithMultipleElementsList() {
+        ReverseKGroup reverser = new ReverseKGroup();
+
+        // Create a list with multiple elements (1 -> 2 -> 3 -> 4 -> 5)
+        Node head = new Node(1);
```

Comment:
```suggestion
        Node head;
        head = new Node(1);
```
I'd write like that to conform to the pattern below